### PR TITLE
[mlrun-kit] Integrate kubeflow pipelines 1.8.3 to full CE

### PR DESCRIPTION
**Pull request by eliyahu77**
### Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove irrelevant fields.]

- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### Description

add pipeline 1.8.3 CE

### CODE REVIEW

File: stable/mlrun-kit/override-full.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ global:
     url: mustprovide
     secretName: secretNameofcontainerregistrymustprovide
 
+
```
Comment (quaark) [inline stable/mlrun-kit/override-full.yaml]: remove empty line

File: stable/mlrun-kit/templates/kubeflow/configmaps/pipeline-install-config.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{{- if .Values.pipelines.enabled -}}
+apiVersion: v1
+#eliyahu
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/configmaps/pipeline-install-config.yaml]: remove

File: stable/mlrun-kit/templates/kubeflow/configmaps/pipeline-install-config.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+data:
+  minioServiceHost: "minio.mlrun.svc.cluster.local"
+  minioServicePort: "9000"
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/configmaps/pipeline-install-config.yaml]: have these as values in the values file

File: stable/mlrun-kit/templates/kubeflow/configmaps/pipeline-install-config.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  ConMaxLifeTime: 120s
+  appName: pipeline
+  appVersion: 1.8.3
+  autoUpdatePipelineDefaultVersion: "true"
+  bucketName: mlrun
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/configmaps/pipeline-install-config.yaml]: have these as values in the values file. And what exactly is `autoUpdatePipelineDefaultVersion`? might want to have that false

File: stable/mlrun-kit/templates/kubeflow/configmaps/pipeline-install-config.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  cacheDb: cachedb
+  cacheImage: gcr.io/google-containers/busybox
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/configmaps/pipeline-install-config.yaml]: have these as values in the values file

File: stable/mlrun-kit/templates/kubeflow/configmaps/pipeline-install-config.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  cacheNodeRestrictions: "false"
+  cronScheduleTimezone: UTC
+  dbHost: mysql
+  dbPort: "3306"
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/configmaps/pipeline-install-config.yaml]: have these as values in the values file

File: stable/mlrun-kit/templates/kubeflow/configmaps/workflow-controller-configmap.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
+{{- if .Values.pipelines.enabled -}}
+apiVersion: v1
+data:
+  artifactRepository: |
+    archiveLogs: true
+    s3:
+      ##eliyahu
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/configmaps/workflow-controller-configmap.yaml]: remove

File: stable/mlrun-kit/templates/kubeflow/configmaps/workflow-controller-configmap.yaml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+      #endpoint: "minio-service.kubeflow:9000"
+      #bucket: "mlpipeline"
+      bucket: "mlrun"
+      insecure: true
+      accessKeySecret:
+        name: mlpipeline-minio-artifact
+        key: accesskey
+      secretKeySecret:
+        name: mlpipeline-minio-artifact
+        key: secretkey
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/configmaps/workflow-controller-configmap.yaml]: make these values as well

File: stable/mlrun-kit/templates/kubeflow/configmaps/workflow-controller-configmap.yaml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  containerRuntimeExecutor: pns
+  executor: |
+    imagePullPolicy: IfNotPresent
+    resources:
+      requests:
+        cpu: 0.01
+        memory: 32Mi
+      limits:
+        cpu: 0.5
+        memory: 512Mi
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/configmaps/workflow-controller-configmap.yaml]: have these as values in the values file

File: stable/mlrun-kit/templates/kubeflow/deployments/metadata-envoy-deployment.yaml [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false"
+      creationTimestamp: null
+      labels:
+        application-crd-id: kubeflow-pipelines
+        component: metadata-envoy
+    spec:
+      containers:
+      - image: gcr.io/ml-pipeline/metadata-envoy:1.8.3
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/metadata-envoy-deployment.yaml]: move to values

File: stable/mlrun-kit/templates/kubeflow/deployments/metadata-grpc-deployment.yaml [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+          valueFrom:
+            configMapKeyRef:
+              key: dbHost
+              name: pipeline-install-config
+        - name: MYSQL_PORT
+          valueFrom:
+            configMapKeyRef:
+              key: dbPort
+              name: pipeline-install-config
+        image: gcr.io/tfx-oss-public/ml_metadata_store_server:1.5.0
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/metadata-grpc-deployment.yaml]: move to values

File: stable/mlrun-kit/templates/kubeflow/deployments/metadata-writer.yaml [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+        application-crd-id: kubeflow-pipelines
+    spec:
+      containers:
+      - env:
+        - name: NAMESPACE_TO_WATCH
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        image: gcr.io/ml-pipeline/metadata-writer:1.8.3
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/metadata-writer.yaml]: move to values

File: stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline-persistenceagent.yaml [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        - name: TTL_SECONDS_AFTER_WORKFLOW_FINISH
+          value: "86400"
+        - name: NUM_WORKERS
+          value: "2"
+        image: gcr.io/ml-pipeline/persistenceagent:1.8.3
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline-persistenceagent.yaml]: move to values

File: stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline-scheduledworkflow.yaml [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        - name: CRON_SCHEDULE_TIMEZONE
+          valueFrom:
+            configMapKeyRef:
+              key: cronScheduleTimezone
+              name: pipeline-install-config
+        image: gcr.io/ml-pipeline/scheduledworkflow:1.8.3
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline-scheduledworkflow.yaml]: move to values

File: stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline-ui.yaml [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+        - name: MINIO_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              key: accesskey
+              name: mlpipeline-minio-artifact
+        - name: MINIO_SECRET_KEY
+          valueFrom:
+            secretKeyRef:
+              key: secretkey
+              name: mlpipeline-minio-artifact
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline-ui.yaml]: move to values

File: stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline-ui.yaml [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+        - name: ALLOW_CUSTOM_VISUALIZATIONS
+          value: "true"
+        image: gcr.io/ml-pipeline/frontend:1.8.3
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline-ui.yaml]: move to values

File: stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline-viewer-crd.yaml [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+    spec:
+      containers:
+      - env:
+        - name: MAX_NUM_VIEWERS
+          value: "50"
+        - name: MINIO_NAMESPACE
+#          valueFrom:
+#            fieldRef:
+#              apiVersion: v1
+#              fieldPath: metadata.namespace
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline-viewer-crd.yaml]: what's going on here?

File: stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline-viewer-crd.yaml [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+        image: gcr.io/ml-pipeline/viewer-crd-controller:1.8.1
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline-viewer-crd.yaml]: move to values

File: stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline-visualizationserver.yaml [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+    metadata:
+      annotations:
+        cluster-autoscaler.kubernetes.io/safe-to-evict: "true"
+      creationTimestamp: null
+      labels:
+        app: ml-pipeline-visualizationserver
+        application-crd-id: kubeflow-pipelines
+    spec:
+      containers:
+      - image: gcr.io/ml-pipeline/visualization-server:1.8.3
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline-visualizationserver.yaml]: move to values

File: stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline.yaml [hunk 20]
```diff
@@ -0,0 +1,152 @@
...
+          valueFrom:
+            configMapKeyRef:
+              key: autoUpdatePipelineDefaultVersion
+              name: pipeline-install-config
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+         #eliyahu
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline.yaml]: remove

File: stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline.yaml [hunk 21]
```diff
@@ -0,0 +1,152 @@
...
+        - name: DBCONFIG_DBNAME
+          valueFrom:
+            configMapKeyRef:
+              key: pipelineDb
+              name: pipeline-install-config
+        - name: DBCONFIG_HOST
+          valueFrom:
+            configMapKeyRef:
+              key: dbHost
+              name: pipeline-install-config
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline.yaml]: move to values

File: stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline.yaml [hunk 22]
```diff
@@ -0,0 +1,152 @@
...
+          valueFrom:
+            secretKeyRef:
+              key: accesskey
+              name: mlpipeline-minio-artifact
+        - name: OBJECTSTORECONFIG_SECRETACCESSKEY
+          valueFrom:
+            secretKeyRef:
+              key: secretkey
+              name: mlpipeline-minio-artifact
+        image: gcr.io/ml-pipeline/api-server:1.8.3
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/ml-pipeline.yaml]: move to values

File: stable/mlrun-kit/templates/kubeflow/deployments/mysql.yaml [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+    spec:
+      containers:
+      - args:
+        - --ignore-db-dir=lost+found
+        - --datadir
+        - /var/lib/mysql
+        env:
+        - name: MYSQL_ALLOW_EMPTY_PASSWORD
+          value: "true"
+        image: gcr.io/ml-pipeline/mysql:5.7-debian
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/mysql.yaml]: move to values

File: stable/mlrun-kit/templates/kubeflow/deployments/workflow-controller.yaml [hunk 24]
```diff
@@ -0,0 +1,89 @@
...
+        - --namespaced
+        command:
+        - workflow-controller
+        env:
+        - name: LEADER_ELECTION_IDENTITY
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        image: gcr.io/ml-pipeline/workflow-controller:v3.3.8-license-compliance
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/deployments/workflow-controller.yaml]: move to values

File: stable/mlrun-kit/templates/kubeflow/secrets/mlpipeline-minio-artifact.yaml [hunk 25]
```diff
@@ -0,0 +1,13 @@
+{{- if .Values.pipelines.enabled -}}
+apiVersion: v1
+data:
+  accesskey: bWluaW8=
+  secretkey: bWluaW8xMjM=
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/secrets/mlpipeline-minio-artifact.yaml]: use base64 on the values from the values file

File: stable/mlrun-kit/templates/kubeflow/secrets/mysql-secret.yaml [hunk 26]
```diff
@@ -0,0 +1,13 @@
+{{- if .Values.pipelines.enabled -}}
+apiVersion: v1
+data:
+  password: ""
+  username: cm9vdA==
```
Comment (quaark) [inline stable/mlrun-kit/templates/kubeflow/secrets/mysql-secret.yaml]: same ^

File: stable/mlrun-kit/values.yaml [hunk 27]
```diff
@@ -137,10 +139,10 @@ mlrun:
...
 
   httpDB:
-    dbType: "sqlite"
-    dirPath: "/mlrun/db"
-    dsn: "sqlite:////mlrun/db/mlrun.db?check_same_thread=false"
-    oldDsn: ""
+    dbType: mysql
+    dsn: mysql+pymysql://root@mlrun-db:3306/mlrun
+    oldDsn: sqlite:////mlrun/db/mlrun.db?check_same_thread=false
+
```
Comment (quaark) [inline stable/mlrun-kit/values.yaml]: Don't do this change yet. There's an issue with the CI that's failing to bring up mysql like this

**Review by quaark — CHANGES_REQUESTED** [review]
Good stuff. Mainly all the hardcoded settings need to be moved to the values file (images and env vars mainly)
Also please rename the dir from `kubeflow` to `pipelines`. those are 2 different things (pipelines is part of kubeflow)

File: stable/mlrun-kit/templates/pipelines/deployments/ml-pipeline-viewer-crd.yaml [hunk 28]
```diff
@@ -0,0 +1,56 @@
...
+    spec:
+      containers:
+      - env:
+        - name: MAX_NUM_VIEWERS
+          value: "50"
+        - name: MINIO_NAMESPACE
+#          valueFrom:
+#            fieldRef:
+#              apiVersion: v1
+#              fieldPath: metadata.namespace
```
Comment (quaark) [inline stable/mlrun-kit/templates/pipelines/deployments/ml-pipeline-viewer-crd.yaml]: Why is this here and empty like this?

File: stable/mlrun-kit/values.yaml [hunk 29]
```diff
@@ -142,6 +144,7 @@ mlrun:
     dsn: "sqlite:////mlrun/db/mlrun.db?check_same_thread=false"
     oldDsn: ""
 
+
```
Comment (quaark) [inline stable/mlrun-kit/values.yaml:147]: remove empty line

**Review by quaark — CHANGES_REQUESTED** [review]
Looks Good!
A couple of minor things.
@liranbg please give a review here as well (keep in mind this will all be moved to the new helm chart repo once I finish working on it)

File: stable/mlrun-kit/templates/pipelines/configmaps/workflow-controller-configmap.yaml [hunk 30]
```diff
@@ -0,0 +1,31 @@
...
+      endpoint: "{{ .Values.pipelines.minio.endpoint }}:{{ .Values.pipelines.minio.endpointPort }}"
+      bucket: "{{ .Values.pipelines.minio.bucket }}"
+      insecure: true
+      accessKeySecret:
+        name: mlpipeline-minio-artifact
+        key: accesskey
+      secretKeySecret:
+        name: mlpipeline-minio-artifact
+        key: secretkey
+  containerRuntimeExecutor: pns
```
Comment (liranbg) [inline stable/mlrun-kit/templates/pipelines/configmaps/workflow-controller-configmap.yaml]:
```suggestion
  containerRuntimeExecutor: emissary
```

File: stable/mlrun-kit/templates/pipelines/configmaps/workflow-controller-configmap.yaml [hunk 31]
```diff
@@ -0,0 +1,31 @@
...
+  executor: |
+    imagePullPolicy: IfNotPresent
+    resources:
+      requests:
+        cpu: 0.01
+        memory: 32Mi
+      limits:
+        cpu: 0.5
```
Comment (liranbg) [inline stable/mlrun-kit/templates/pipelines/configmaps/workflow-controller-configmap.yaml:24]: low, isn't it?

**Review by liranbg — CHANGES_REQUESTED** [review]
minor change to align with new kubeflow

**Review by quaark — APPROVED** [review]
LGTM, please address @liranbg's comments and update the chart version to `0.5.0-rc.11`.

File: stable/mlrun-kit/templates/pipelines/configmaps/workflow-controller-configmap.yaml (hunk 31 above)
Comment (eliyahu77) [reply stable/mlrun-kit/templates/pipelines/configmaps/workflow-controller-configmap.yaml:24]: lest keep it for now  with this values if needed we will increase

**Review by liranbg — APPROVED** [review]
